extensions/wikia/GoogleAmp/GoogleAmpHelper.class.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+class GoogleAmpHelper {
+
+	/**
+	 * Checks whether AMP version is enabled for a given title.
+	 *
+	 * @param Title $title
+	 * @return bool
+	 */
+	static public function isAmpEnabled( Title $title ): bool {
+		global $wgGoogleAmpNamespaces, $wgGoogleAmpArticleBlacklist;
+
+		if ( in_array( $title->getNamespace(), $wgGoogleAmpNamespaces ) ) {
+			if ( !in_array( $title->getPrefixedDBkey(), $wgGoogleAmpArticleBlacklist ) ) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * If Amp is enabled for a given title, return Amp article address.
+	 *
+	 * @param Title $title
+	 * @return string|null
+	 */
+	static public function getAmpAddress( Title $title ) {
+		global $wgServer, $wgGoogleAmpAddress;
+
+		if ( !self::isAmpEnabled( $title ) ) {
+			return null;
+		}
+
+		$wikiServer = WikiFactory::getLocalEnvURL( $wgServer, WIKIA_ENV_PROD );
+		$serverRegex = '/^https?:\/\/(.+)\.wikia\.com/';
+		if ( preg_match( $serverRegex, $wikiServer, $groups ) !== 1 ) {
+			return null;
+		}
+		$wikiServer = rawurlencode( $groups[1] );
+		$article = rawurlencode( $title->getPrefixedDBkey() );
```

Comment:
How about `$article = $title->getPartialURL()`?

Reply:
`getPartialURL` doesn't have the NS prefix, so for `wiki/Aaa` and `wiki/User:Aaa` produces the same result.